kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/ClientConsumerRecord.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.apache.kafka.common.header.internals.RecordHeaders;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The type Client consumer record.
+ */
+public class ClientConsumerRecord extends BaseConsumerRecord {
```

Comment:
I'd use Java records for these classes.

Reply:
We refused to use Java as name for Strimzi test clients, but renamed to StrimziTestClientConsumerRecord

Reply:
Ok.. But I actually meant `public record ClientConsumerRecord implements ConsumerRecord`

But I'm happy enough with your ConsumerRecord class.